File: src/main/java/ru/example/configuration/RepositoryConfiguration.java
package ru.example.configuration;

import org.springframework.context.annotation.Bean;
import org.springframework.data.rest.webmvc.config.RepositoryRestConfigurer;
import ru.example.model.Author;
import ru.example.model.Book;
import ru.example.model.Tag;

public class RepositoryConfiguration {

    @Bean
    public RepositoryRestConfigurer repositoryConfig() {
        return RepositoryRestConfigurer.withConfig(
                repositoryRestConfiguration -> {
                    repositoryRestConfiguration.exposeIdsFor(
                            Book.class,
                            Tag.class,
                            Author.class
                    );
                });
    }
}

File: src/main/java/ru/library/configuration/RepositoryConfiguration.kt
package ru.library.configuration

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.data.rest.core.config.RepositoryRestConfiguration
import org.springframework.data.rest.webmvc.config.RepositoryRestConfigurer
import ru.library.model.Author
import ru.library.model.Book
import ru.library.model.Tag

@Configuration
class RepositoryConfiguration {
    @Bean
    fun repositoryConfig(): RepositoryRestConfigurer {
        return RepositoryRestConfigurer.withConfig { repositoryRestConfiguration: RepositoryRestConfiguration ->
            repositoryRestConfiguration.exposeIdsFor(
                    Book::class.java,
                    Tag::class.java,
                    Author::class.java
            )
        }
    }
}